### PR TITLE
app-content-pages: Built a shared sticky side bar component

### PR DESCRIPTION
**Pull request by goplayoutside3**
~~⚠️ This contains #5488 and #5551 please review those first~~

## Package
app-content-pages

## Linked Issue and/or Talk Post
Toward: https://github.com/zooniverse/front-end-monorepo/issues/5476
Project board: https://github.com/orgs/zooniverse/projects/40
Figma design: https://www.figma.com/file/zN9InDg8CTk2vnWp406moK/About-Pages

## Describe your changes

- The new designs for the About Zooniverse pages include a "scroll to" UX rather than filtered data. I refactored the side bar links to bring a user to the linked section (publications category or team location) rather than filter the data and re-render. The side bar is now sticky, too.
- I extracted those side bar links to a Sidebar component because the same design is used in the root About page (to be built). I added a story and unit tests for that Sidebar.
- I removed the TwoColumnLayout component in favor of styling the Teams and Publications pages directly in their files.
- I removed the TeamsContainer and PublicationsContainer files because their filtering logic is no longer needed

## How to Review

- Run app-content-pages storybook and view the new Sidebar story
- Run app-content-pages locally and check out https://local.zooniverse.org:3000/about/publications and https://local.zooniverse.org:3000/about/teams. Click the side bar to scroll to a section
- I would like to explore the Intersection Observer API for detecting when a user has reached a certain section while scrolling down, and adjust the active section link in the sidebar (and the url?)
- There may be a few design adjustments - I'll tag Sean here with a couple of a UI questions

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## New Feature
- [ ] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by goplayoutside3** [comment]
@seanmiller26 here's an example screenshot of an active link after I clicked on "Minnesota". The link takes me to that section of the page. 

Screenshot 2023-10-30 at 9 08 50 PM

UI question - Do you think the side bar should include an "All" or "Top of Page" link? This might be helpful especially on the Publications page. When someone wants to return to the top of the page, they could click Space because it's the first category, but I'm not sure that's intuitive.

File: packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  return (
+    // <Nav aria-label={t('Publications.sideBarLabel')} className={className}>
+    <Nav aria-label={ariaLabel} className={className}>
+      <StyledUl>
+        {sections.map(section => (
+          <StyledLi key={section.name}>
+            <StyledButton
+              as={Link}
+              active={section.slug === activeSection}
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js:52]:
```suggestion
              active={section.slug === activeSection}
              aria-current = section.slug === activeSection ? 'true' : 'false'
```
`aria-current` here would provide a hint to screen readers as to which link is active.

**Review by eatyourgreens — APPROVED** [review]
Looks good. There's a slight flicker of the sidebar when the page first loads, because the active link is set by `useEffect`. The current selection in a list of links should also be indicated with [`aria-current`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current).

**Comment by seanmiller26** [comment]
@goplayoutside3 That is a very intuitive solution. +1 for including "All" at the top of the sidebar menu for Team and Publications page.

File: packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/shared/components/Sidebar/Sidebar.js:52]: Good point - thanks!

**Comment by goplayoutside3** [comment]
> ...including "All" at the top of the sidebar menu

Sounds good, I added a link labeled "All" to both the Teams and Publications page. There are no other translation dictionaries solidified for this content yet, so no problem if we want to change that label string during development of the new About Zooniverse designs.

**Pull request merged**